packages/xo-server/src/api/vm.js

```diff
@@ -480,24 +482,29 @@ export async function migrate({
 
   await this.checkPermissions(this.user.id, permissions)
 
-  await this.getXapi(vm).migrateVm(
-    vm._xapiId,
-    this.getXapi(host),
-    host._xapiId,
-    {
+  await this.getXapi(vm)
+    .migrateVm(vm._xapiId, this.getXapi(host), host._xapiId, {
       sr: sr && this.getObject(sr, 'SR')._xapiId,
       migrationNetworkId:
         migrationNetwork != null ? migrationNetwork._xapiId : undefined,
       mapVifsNetworks: mapVifsNetworksXapi,
       mapVdisSrs: mapVdisSrsXapi,
-    }
-  )
+      force,
+    })
+    .catch(error => {
+      if (error.code === 'VM_INCOMPATIBLE_WITH_THIS_HOST') {
```

Comment:
`error` could be `undefined` or `null`, in that case, it will throw `TypeError: Cannot read property 'code' of undefined` which will hide the original error.

```suggestion
      if (error?.code === 'VM_INCOMPATIBLE_WITH_THIS_HOST') {
```